### ISSUE TITLE
long filenames error when using compcor

### ISSUE THREAD

**Issue by davidreydellet** [CPernet/spmup#44]
Hello,

I used spmup applied to BIDS data with 2 different sets of options. It works well with the 1st set and I may have identified a bug when using compcor as in my 2nd set of options. 

I get this warning:
_Warning: 'sub_d1s01_ses_lsd_task_rest_acq_ep2d_space_MNI152NLin2009_desc_preprocessed_bold_design' exceeds MATLAB's maximum name length of 63 characters and has been truncated to 'sub_d1s01_ses_lsd_task_rest_acq_ep2d_space_MNI152NLin2009_desc_'._ 

Followed by these errors:
_Error using fileparts (line 37)
Input must be a row vector of characters or string scalar._

_Error in spmup_BIDS_1rstlevel (line 256)
            [filepath,filename,ext]=fileparts(matlabbatch{1}.spm.stats.fmri_spec.sess(frun).scans);_

_Error in run_spmup_bids (line 169)
                    subject_sess = spmup_BIDS_1rstlevel(subject_sess, opt(s));_

I tried to shorten the filenames to see whether the issue really comes from this but I did not manage to get under this 63 characters cap.

Here is a mock dataset to reproduce this error (with two scripts: one with a set of options that works fine, and one with a set of options yielding this error): /users/davidreydellet/mock_datasets/

**Comment by Remi-Gau** [comment]
hey @davidreydellet 
was there supposed to be a link to some data and code ?

**Comment by Remi-Gau** [comment]
MATLAB is pretty strict as to what can be a valid variable or field name: can be an interoperability issue when creating tsv headers or JSON fieldnames that may work in python but lead to "fun" things with matlab.

https://bids-standard.github.io/bids-starter-kit/folders_and_files/metadata.html#interoperability-issues

**Comment by CPernet** [comment]
so error is unrelated to the warning - you can see the issue is when parsing the scan name
@davidreydellet are you ok with a bit of coding? you could put a debug point in https://github.com/CPernet/spmup/blob/master/bids/run_spmup_bids.m#L168 which means the processing is done
then check subject_sess it should contains the right scan names in, but likely not hence the bug ; then report to us and we'll see what we can do to help you from there 

thx for testing the tool

**Comment by davidreydellet** [comment]
Thank you for developing this tool and following its implementation!

I checked a bit, at this stage (run_spmup_bids L168), the variable subject_sess does contain filenames but I presume some are missing. In my case, my raw data comes with T1w and T2w images, fieldmaps, and a functional data file. I will deep dive a bit into spmup_BIDS_1rstlevel.

[subject_sess.mat.tar.gz](https://github.com/CPernet/spmup/files/10684232/subject_sess.mat.tar.gz)

**Comment by CPernet** [comment]
ok but it works in other BIDS datasets we tried, so we need to figure out why in your case, as this is not a tenable solution ; let me check again with the file your shared

**Comment by davidreydellet** [comment]
The first 7 subjects of my dataset got processed with this modification but an error appeared at subject#7. I will try to sort out this error now.

```
------------------------------
 running spmup_timeseries on /data1/lsd-basel/data/derivative/dataset1/sub-d1s07/ses-lsd/func/run1/sub-d1s07_ses-lsd_task-rest_acq-ep2d_space-MNI152NLin2009_desc-preprocessed-GLMdenoised_bold.nii
------------------------------
Checking Yeo's networks to organize GM voxels
Error using corr (line 91)
Requires a data matrix X.

Error in spmup_timeseriesplot (line 297)
    tmp                              = corr(tmp);

Error in spmup_BIDS_1rstlevel (line 263)
        spmup_timeseriesplot(VfMRI(1).fname, ...

Error in run_spmup_bids (line 169)
                    subject_sess = spmup_BIDS_1rstlevel(subject_sess, opt(s));
```

**Comment by CPernet** [comment]
x likely empty??

**Comment by davidreydellet** [comment]
Indeed, in this loop going through all ROIs, when it comes to roi 5, tmp is filled with NaNs and subsequently becomes 0, hence the error with the corr function.
From what I understood, roi 5 is the limbic network in Yeo's 7 networks atlas. I guess it could be that this region in particular is absent from my functional data. I visually checked the functional data, it seems ok but maybe my visual check is not enough at this stage. I noticed this error with another subject too.

In all cases, this seems like an error that comes from my data itself rather than the code so I don't want to mix things up in this thread.

To come back to the initial error, the `char(matlabbatch{1}.spm.stats.fmri_spec.sess(frun).scans)` implementation works for me but I agree it's rather peculiar that spmup worked on bids datasets without this modification.

**Comment by CPernet** [comment]
@davidreydellet can you pull the updated version and see if setting options.reorient   = 'off'; solves some of the problem
@Remi-Gau options.reorient   = 'off'; goes around issue https://github.com/CPernet/spmup/issues/5 while ultimately we need a better code to reorient for defaced data
